components/builder-vault/src/config.rs

```diff
@@ -62,8 +70,24 @@ impl ConfigFile for Config {
     fn from_toml(toml: toml::Value) -> Result<Self> {
         let mut cfg = Config::default();
         try!(toml.parse_into("cfg.routers", &mut cfg.routers));
-        try!(toml.parse_into("cfg.datastore_addr", &mut cfg.datastore_addr));
-        try!(toml.parse_into("cfg.datastore_retry_ms", &mut cfg.datastore_retry_ms));
+
+        let mut connection_user = String::new();
+        try!(toml.parse_into("cfg.datastore_connection_user", &mut connection_user));
+        let mut connection_address = String::new();
+        try!(toml.parse_into("cfg.datastore_connection_address", &mut connection_address));
+        let mut connection_db = String::new();
+        try!(toml.parse_into("cfg.datastore_connection_db", &mut connection_db));
+
+        cfg.datastore_connection_url = format!("postgresql://{}@{}/{}",
+                                               connection_user,
+                                               connection_address,
+                                               connection_db);
+        try!(toml.parse_into("cfg.datastore_connection_retry_ms",
+                             &mut cfg.datastore_connection_retry_ms));
+        let mut timeout_seconds = 0;
```

Comment:
This should be initialized to 3600 (default value), not 0, since we want a sane value even if we are parsing only a partial set of config values.